### PR TITLE
Added tests to guide in addressing issue #478.

### PR DESCRIPTION
**Pull request by afd**
Tests to guide in addressing issue #478.

Search for TODO(478) in the code; there are three @Ignored tests.

Once the issue is fixed, these test should be enabled.

### CODE REVIEW

File: reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/RemoveRedundantUniformMetadataReductionOpportunitiesTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 The GraphicsFuzz Project Authors
```
Comment (paulthomson) [inline reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/RemoveRedundantUniformMetadataReductionOpportunitiesTest.java]: Year

File: reducer/src/test/java/com/graphicsfuzz/reducer/ReductionDriverTest.java [hunk 2]
```diff
@@ -561,6 +563,44 @@ public void testReductionWithUniformBindings() throws Exception {
...
+    final String resultsPrefix = new ReductionDriver(new ReducerContext(true,
+        ShadingLanguageVersion.ESSL_100,
+        new RandomWrapper(0),
+        new IdGenerator(), true),
+        false,
+        fileOps,
+        (unused, item) -> true, workDir)
+        .doReduction(shaderJob, "temp", 0, 100);
+
+    final ShaderJob after = fileOps.readShaderJobFile(new File(testFolder.getRoot(),resultsPrefix + ".json"));
```
Comment (paulthomson) [inline reducer/src/test/java/com/graphicsfuzz/reducer/ReductionDriverTest.java]: Nit: comma has no space

File: reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/RemoveRedundantUniformMetadataReductionOpportunitiesTest.java (hunk 1 above)
Comment (afd) [reply reducer/src/test/java/com/graphicsfuzz/reducer/reductionopportunities/RemoveRedundantUniformMetadataReductionOpportunitiesTest.java]: Fixed

File: reducer/src/test/java/com/graphicsfuzz/reducer/ReductionDriverTest.java (hunk 2 above)
Comment (afd) [reply reducer/src/test/java/com/graphicsfuzz/reducer/ReductionDriverTest.java]: Fixed.

Checkstyle doesn't run on tests by default.  Perhaps we should make it do so (we would have to fix up all the tests but that shouldn't take long.)

**Pull request merged**